sdk/event/event_subscription_plugin.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package event
+
+import (
+	"context"
+)
+
+type EventSubscriptionPlugin interface {
+	Initialize(ctx context.Context) error
+	Subscribe(ctx context.Context, request *SubscribeRequest) error
+	SendSubscriptionEvent(subscriptionID string, eventJson string) error
+	Unsubscribe(ctx context.Context, subscriptionID string) error
+	// Type returns the name and version for the particular event subscription plugin.
```

Comment:
For secrets and auth plugins, `Type` returns "auth" or "secret" - might be nice to find a different name that avoids confusing the two, as the codebase already has a lot of confusion about what a "type" is wrt plugins.

Reply:
Oh, that is confusing. Maybe I'll change it to `PluginVersion()`, and it can contain the name and version?